core/src/test_utils.rs

```diff
@@ -14,19 +14,27 @@ where
 #[macro_export]
 macro_rules! assert_events {
     ($contract:ident, $ ( $event_ty:ty ),+ ) => {
-        let mut __idx = 0;
+        let mut __idx = -1;
         $(
             __idx -= 1;
+            let _ = $event_ty;
```

Comment:
What is this?  What's more, it does not compile. In the other case, it works fine because, in the second arm, we've got an expression so having `Transfer` event the let statement may look like `let _ = Transfer { .. }` but here is `let _ = Transfer`